### DISCUSSION TITLE
Should browser compatibility be disregarded? 是否应当不考虑浏览器兼容性？

### DISCUSSION THREAD

**Discussion by TheNovemberRain** [element-plus/element-plus#15569] [Polls]
See this PR for details: [#15313](https://github.com/element-plus/element-plus/pull/15313)

The big reason I would use element plus over other component libraries is that it clearly defines the minimum supported version, and it's the version that the vast majority of computers are running. I think it would be very irresponsible to say "last two versions supported" as other libraries do. Developers don't know what features are not supported, and it's not clear if element plus will work as expected with the addition of Babel and Polyfill. I hope it would be more appropriate to raise the minimum supported version to a specific version number, like 99.

我会使用element plus而不是其他组件库的重要原因是它明确定义了最低支持的版本，而且是绝大多数电脑都在运行的版本。如果像其他组件库一样标识“支持最后两个版本”，我认为是非常不负责任的。开发者并不清楚哪些特性是不受支持的，也不清楚在添加Babel和Polyfill后，element plus是否还能如期正常运作。我希望可以提高最低支持的版本到一个特定的版本号，比如99，这样会更加合适。

**Comment by kooriookami** [comment]
It makes sense. We'll mark the supported browsers version in each minor version.

**Reply by TheNovemberRain** [reply]
Element Plus has always done a great job with browser compatibility, and there is no other PC component library like it.
However, this PR does not specify the browser versions supported by version 2.5.0. Is it possible to give a detailed migration plan? Preferably contains new features used in the current version, including js and css.

For example:

**2.5.x**
Supported browsers: chrome 85、firefox xxx、xxx
Migration from 2.4.x:
css:
1. [flexbox-gap](https://caniuse.com/flexbox-gap): [flex-gap-polyfill](https://github.com/gavinmcfarland/flex-gap-polyfill)
2. xxx: xxx

javascript:
1. [Web Animations API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API): [web-animations-js](https://github.com/web-animations/web-animations-js)
2. xxx: xxx

**2.8.x**
supported browsers: chrome 102、firefox xxx、xxx
Migration from 2.7.x:
css:
1. xxx: xxx
2. xxx: xxx

javascript:
1. xxx: xxx
2. xxx: xxx

This leads to painless updates where developers know exactly which features can be downgraded and which features cannot be downgraded if upgrading from an older version to a newer one.

In some scenarios, updating the browser version is difficult and developers have to consider the lowest supported version. For example:
1. the presence of thousands of computers in LAN environments without Internet, where their browsers are uniformly distributed, making updates costly.
2. medical systems running on mobile devices in remote areas with weak network environments.

I think ToB systems require a high level of browser compatibility, so I would recommend exercising restraint on the matter of increasing the minimum supported browser version, and not using new features if you can.

Thank you.
